### PR TITLE
removed random tests

### PR DESCRIPTION
**Pull request by jochman**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/29251

## Description
Ne
## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Demisto
- [ ] 5.0.0
- [ ] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

### CODE REVIEW

File: Tests/scripts/collect_tests_and_content_packs.py [hunk 1]
```diff
@@ -1183,10 +1170,7 @@ def get_test_list_and_content_packs_to_install(files_string, branch_name, minimu
         tests = tests.union(get_test_from_conf(branch_name, conf))
 
     if not tests:
-        rand = random.Random(branch_name)
-        tests = get_random_tests(
-            tests_num=RANDOM_TESTS_NUM, rand=rand, conf=conf, id_set=id_set, server_version=minimum_server_version)
-        packs_to_install = get_content_pack_name_of_test(tests, id_set)
+        packs_to_install = set()  # No random tests
         if changed_common:
             logging.debug('Adding 3 random tests due to: {}'.format(','.join(changed_common)))
```
Comment (Itay4) [inline Tests/scripts/collect_tests_and_content_packs.py:1175]: need to update this log

**Comment by Itay4** [comment]
what tests do we have for this? did we trigger a build against these changes?

**Comment by hod-alpert** [comment]
That's not enough IMO.
As written in the issue https://github.com/demisto/etc/issues/29251
We need four playbooks that will be run instead of the random tests and the sample test stated by the quoted code:
```
tests.add('TestCommonPython')  # test with no integration configured
tests.add('HelloWorld-Test')  # test with integration configured
packs_to_install.add("HelloWorld")
```
We need to write 4 **new** playbooks that will be as **light and fast** as possible with the following criteria:
Playbook without integration - Pure sanity test for the infra.
Playbook with integration - Making sure the build infra can configure an integration with the configuration from `content-test-conf`.
Playbook with mock - Making sure that the MITMproxy mechanism is working.
Playbook without mock - Making sure that the locking mechanism is working.

The main goal of these playbooks will be to check the infrastructure of the build so these tests should not do much, even one or two steps in each will be enough

File: Tests/scripts/collect_tests_and_content_packs.py [hunk 2]
```diff
@@ -1183,16 +1170,8 @@ def get_test_list_and_content_packs_to_install(files_string, branch_name, minimu
...
-        tests = get_random_tests(
-            tests_num=RANDOM_TESTS_NUM, rand=rand, conf=conf, id_set=id_set, server_version=minimum_server_version)
-        packs_to_install = get_content_pack_name_of_test(tests, id_set)
-        if changed_common:
-            logging.debug('Adding 3 random tests due to: {}'.format(','.join(changed_common)))
-        elif sample_tests:  # Choosing 3 random tests for infrastructure testing
-            logging.debug('Collecting sample tests due to: {}'.format(','.join(sample_tests)))
-        else:
-            logging.debug("Running Sanity check only")
+        packs_to_install = set()  # No random tests
```
Comment (hod-alpert) [inline Tests/scripts/collect_tests_and_content_packs.py:1173]:
```suggestion
        packs_to_install = set()
```

File: Tests/scripts/collect_tests_and_content_packs.py [hunk 3]
```diff
@@ -1183,16 +1170,8 @@ def get_test_list_and_content_packs_to_install(files_string, branch_name, minimu
...
+        logging.debug("Running Sanity check only")
```
Comment (hod-alpert) [inline Tests/scripts/collect_tests_and_content_packs.py:1174]:
```suggestion
        logging.info("No tests found running sanity check only")
```

File: Tests/scripts/collect_tests_and_content_packs.py [hunk 4]
```diff
@@ -1183,16 +1170,8 @@ def get_test_list_and_content_packs_to_install(files_string, branch_name, minimu
...
 
         tests.add('TestCommonPython')  # test with no integration configured
         tests.add('HelloWorld-Test')  # test with integration configured
```
Comment (hod-alpert) [inline Tests/scripts/collect_tests_and_content_packs.py:1177]:
```suggestion
        logging.debug('Adding test playbook with no integration configured: TestCommonPython')
        tests.add('TestCommonPython')
        logging.debug('Adding test playbook with integration configured: HelloWorld-Test')
        tests.add('HelloWorld-Test')
```

**Review by hod-alpert — APPROVED** [review]
Just fix these comments and add another PR with the new test playbooks

File: Tests/scripts/collect_tests_and_content_packs.py [hunk 5]
```diff
@@ -1183,19 +1170,12 @@ def get_test_list_and_content_packs_to_install(files_string, branch_name, minimu
...
-        tests = get_random_tests(
-            tests_num=RANDOM_TESTS_NUM, rand=rand, conf=conf, id_set=id_set, server_version=minimum_server_version)
-        packs_to_install = get_content_pack_name_of_test(tests, id_set)
-        if changed_common:
-            logging.debug('Adding 3 random tests due to: {}'.format(','.join(changed_common)))
-        elif sample_tests:  # Choosing 3 random tests for infrastructure testing
-            logging.debug('Collecting sample tests due to: {}'.format(','.join(sample_tests)))
-        else:
-            logging.debug("Running Sanity check only")
+        packs_to_install = set()
```
Comment (yaakovi) [inline Tests/scripts/collect_tests_and_content_packs.py:1173]: redundant, already initialized at [line 1167](https://github.com/demisto/content/pull/9806/files#diff-25b80cc78a0efaee708db1ce39fb154ac3358eae6361e3e336922d8a77599863R1167)

**Review by yaakovi — COMMENTED** [review]
looks ok to me

**Comment by hod-alpert** [comment]
Rename the playbooks, the current names are unclear I think the names should be:
1. `Sanity Test - Playbook with no integration` (instead of the Pure one)
2. `Sanity Test - Playbook with mocked integration` (instead of the configuration one)
3. You can remove the MITMproxy one since the configuration one uses HelloWorld command too)
4. Add another test playbook that uses an unmockable integration like Gmail and list emails with limit of 3

**Review by hod-alpert — CHANGES_REQUESTED** [review]
Rename the playbooks, the current names are unclear I think the names should be:
1. `Sanity Test - Playbook with no integration` (instead of the Pure one)
2. `Sanity Test - Playbook with mocked integration` (instead of the configuration one)
3. You can remove the MITMproxy one since the configuration one uses HelloWorld command too which is a mocked integration)
4. Add another test playbook that uses an unmockable integration like Gmail and list emails with limit of 3
Lets call it `Sanity Test - Playbook with unmocked integration`

**Comment by jochman** [comment]
> 2\. Sanity Test - Playbook with mocked integration

The HelloWorld one isn't using any API call

**Comment by hod-alpert** [comment]
> > 2. Sanity Test - Playbook with mocked integration
> 
> The HelloWorld one isn't using any API call

Got you, So remove the configuration one

File: Packs/HelloWorld/TestPlaybooks/Sanity_Test_-_MITMProxy.yml [hunk 6]
```diff
@@ -0,0 +1,218 @@
+id: Sanity Test - MITMProxy
```
Comment (hod-alpert) [inline Packs/HelloWorld/TestPlaybooks/Sanity_Test_-_MITMProxy.yml:1]: Rename it to `Sanity Test - Playbook with mocked integration`

File: Packs/HelloWorld/TestPlaybooks/Sanity_Test_-_Mocked_Integration.yml [hunk 7]
```diff
@@ -0,0 +1,139 @@
+id: Sanity Test - Playbook with mocked integration
```
Comment (hod-alpert) [inline Packs/HelloWorld/TestPlaybooks/Sanity_Test_-_Mocked_Integration.yml:1]:
```suggestion
id: Sanity Test - Playbook with integration
```

Comment (hod-alpert) [reply Packs/HelloWorld/TestPlaybooks/Sanity_Test_-_Mocked_Integration.yml:1]: Rename the playbook - this does not run any mock

File: Tests/scripts/collect_tests_and_content_packs.py [hunk 8]
```diff
@@ -1043,20 +1029,18 @@ def get_test_list_and_content_packs_to_install(files_string, branch_name, minimu
...
-
-        tests.add('TestCommonPython')  # test with no integration configured
-        tests.add('HelloWorld-Test')  # test with integration configured
+        logging.info("No tests found running sanity check only")
+
+        sanity_tests = {
+            "Sanity Test - Playbook with no integration",
+            "Sanity Test - Playbook with mocked integration",
+            "Sanity Test - MITMProxy",
+            "Sanity Test - Non-Mocked Integration"
```
Comment (hod-alpert) [inline Tests/scripts/collect_tests_and_content_packs.py:1038]:
```suggestion
            "Sanity Test - Playbook with Unmockable Integration"
```

File: Tests/scripts/collect_tests_and_content_packs.py [hunk 9]
```diff
@@ -1043,20 +1029,18 @@ def get_test_list_and_content_packs_to_install(files_string, branch_name, minimu
...
+        }
+        logging.debug(f"Adding sanity tests: {sanity_tests}")
+        tests.update(sanity_tests)
         packs_to_install.add("HelloWorld")
+        packs_to_install.add("Gmail")
```
Comment (hod-alpert) [inline Tests/scripts/collect_tests_and_content_packs.py:1043]: Add a debug log that explains why these packs are being added

**Review by hod-alpert — CHANGES_REQUESTED** [review]
Add the new playbooks to conf.json so that they will run in the build

**Review by hod-alpert — APPROVED** [review]

**Pull request merged**